spec/services/create_next_edition_service_spec.rb

```diff
@@ -0,0 +1,121 @@
+    let(:user) { create :user }
+
+    before do
+      populate_default_government_bulk_data
+    end
+
+    it "aborts if the current edition isn't live" do
+      current_edition = create(:edition, live: false)
+
+      expect { described_class.call(current_edition: current_edition, user: user) }
+        .to raise_error("Can only create a next edition from a live edition")
+    end
+
+    it "returns a new current edition" do
+      current_edition = create(:edition, :published)
+
+      next_edition = described_class.call(current_edition: current_edition,
+                                          user: user)
+
+      expect(current_edition.current).to be(false)
+      expect(next_edition.current).to be(true)
+    end
+
+    it "returns a draft edition" do
+      current_edition = create(:edition, :published)
+
+      next_edition = described_class.call(current_edition: current_edition,
+                                          user: user)
+
+      expect(next_edition.draft?).to be(true)
+    end
+
+    it "resets change_note, update_type and propoposed_publish_time values" do
+      current_edition = create(:edition,
+                               :published,
+                               change_note: "note",
+                               update_type: "minor",
+                               proposed_publish_time: Time.zone.now)
+
+      next_edition = described_class.call(current_edition: current_edition,
+                                          user: user)
+
+      expect(next_edition.change_note).to be_empty
+      expect(next_edition.update_type).to eq("major")
+      expect(next_edition.proposed_publish_time).to eq(nil)
+    end
+
+    it "appends the change note details when there is a change note and a major change" do
+      current_edition = create(:edition,
+                               :published,
+                               number: 2,
+                               published_at: Date.yesterday.noon,
+                               change_note: "note",
+                               update_type: "major")
+
+      next_edition = described_class.call(current_edition: current_edition,
+                                          user: user)
+
+      expect(next_edition.change_history.first).to match a_hash_including(
+        "note" => "note",
+        "public_timestamp" => Date.yesterday.noon.rfc3339,
+      )
+    end
+
+    it "does not update the change history when the current edition is not a major change" do
+      current_edition = create(:edition,
+                               :published,
+                               number: 2,
+                               published_at: Date.yesterday.noon,
+                               change_note: "note",
+                               update_type: "minor")
+
+      next_edition = described_class.call(current_edition: current_edition,
+                                          user: user)
+
+      expect(next_edition.change_history).to be_empty
```

Comment:
should this actually be ` expect(next_edition.change_history).to eq(current_edition.change_history)` as I guess we don't care that it's empty more that it matches.

Reply:
See previous comment.